pkg/controller/machine/controller.go

```diff
@@ -177,43 +177,46 @@ func (r *ReconcileMachine) Reconcile(ctx context.Context, request reconcile.Requ
 	originalConditions := conditions.DeepCopyConditions(m.Status.Conditions)
 
 	if r.gate.Enabled(featuregate.Feature(openshiftfeatures.FeatureGateMachineAPIMigration)) {
-		// Check Status.AuthoritativeAPI
-		// If not MachineAPI. Set the paused condition true and return early.
-		//
-		// Once we have a webhook, we want to remove the check that the AuthoritativeAPI
-		// field is populated.
 		if m.Status.AuthoritativeAPI != "" &&
 			m.Status.AuthoritativeAPI != machinev1.MachineAuthorityMachineAPI {
 			conditions.Set(m, conditions.TrueConditionWithReason(
 				PausedCondition,
 				PausedConditionReason,
-				"The AuthoritativeAPI is set to %s", string(m.Status.AuthoritativeAPI),
+				"The AuthoritativeAPI is set to '%s'", string(m.Status.AuthoritativeAPI),
 			))
-			if patchErr := r.updateStatus(ctx, m, ptr.Deref(m.Status.Phase, ""), nil, originalConditions); patchErr != nil {
+			if patchErr := r.updateStatus(ctx, m, ptr.Deref(m.Status.Phase, ""), nil, originalConditions); patchErr != nil && !apierrors.IsNotFound(patchErr) {
 				klog.Errorf("%v: error patching status: %v", machineName, patchErr)
+				return reconcile.Result{}, fmt.Errorf("error patching status: %w", patchErr)
+			} else if apierrors.IsNotFound(patchErr) {
+				return reconcile.Result{}, nil
```

Comment:
Why are we special casing this? Does it happen a lot? This would suggest that something else has deleted the machine before we got here, which is possible but:
* it seems unlikely
* just throwing the error would be safe because we'd re-reconcile and get the 'normal' object deleted behaviour

If this does happen a lot for some reason, could we have a comment here explaining why? Otherwise I'm in danger of getting stuck down this rabbit hole every time I see this code.

Reply:
Got rid of it